R/mle-finder.R

```diff
@@ -21,3 +21,76 @@ linear_BFGS <- function(design, outcome, noise_var = 1) {
   return(MLE)
 }
 
+logit_loglik <- function(coef, design, outcome) {
+
+  n <- length(outcome)
+  log_lik <- sum(sapply(1:n, function(i){
+    outcome[i] * sum(design[i,] * coef) -
+      log(1 + exp(sum(design[i,] * coef)))
+    }))
+  return(log_lik)
+
+}
+
+logit_gradient <- function(coef, design, outcome) {
+
+  pi <- exp(design %*% coef) / (1 + exp(design %*% coef))
```

Comment:
-2

you're doing `design %*% coef` twice here, bettter to compute once and store. Additionally, you could write an `expit()` function which will be easier to read and reuse each time you want to compute the probabilities. Same for the hessian function